### ISSUE TITLE
Virtual category "Only discounted products" is not working with configurable products

### ISSUE THREAD

**Issue by lukutisz** [Smile-SA/elasticsuite#2847]
### Preconditions

Magento Version : CE 2.4.5-p1

ElasticSuite Version : 2.10.12

Environment : Developer/Production

Third party modules : No

### Steps to reproduce

1. Create a configurable product
2. Add size variants with special price applied
3. Create virtual category with filter "Only configurable products" and "Only discounted products"

### Expected result

1. Configurable products with special price applied appearing in that virtual category

### Actual result

1. No configurable products where special price is applied
2. Only those products appear, where catalog rule applied (not fully sure, but suspect that still not all of these products are visible)

In Magento 2.4.5-p1 and ElasticSuite 2.10.12

This workaround is not working https://github.com/Smile-SA/elasticsuite/pull/2677 , I am getting errors:

`Type Error occurred when creating object: Smile\ElasticsuiteCatalog\Model\Product\Indexer\Fulltext\Datasource\PriceData, Smile\ElasticsuiteCatalog\Model\Product\Indexer\Fulltext\Datasource\PriceData::__construct(): Argument #2 ($attributeResourceModel) must be of type Smile\ElasticsuiteCatalog\Model\ResourceModel\Product\Indexer\Fulltext\Datasource\AttributeData, array given, called in /home/s/public_html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php on line 121`

So right now in virtual categories by selecting Only discounted products we are not getting configurable products, so we do not have ability to have a virtual category where all discounted products would appear correctly. It was mentioned in several PRs and Issues, but nowhere seen to be fully solved.

**Comment by romainruaud** [comment]
Duplicate of #866 

The workaround given in #2677 is the way to go, but yes, you're right, this workaround should be updated to fit with latests Elasticsuite version.

**Comment by romainruaud** [comment]
It's just adding a new parameter in construct as per with the latest version.

**Comment by barbarosselimbuyukelci** [comment]
Hello Romain, the version is 2.10.11. 

It is really so important for us to have a fix for that so if you need anything from me, please do let me know.

**Comment by romainruaud** [comment]
I merged the workaround of Bashev in 2.10.x branch, you can try to use it locally.

Regards

**Comment by cptX** [comment]
Hi, I just hit the same problem and i'm really urged to find a solution. @romainruaud are you sure is a duplicate of https://github.com/Smile-SA/elasticsuite/issues/866?
https://github.com/Smile-SA/elasticsuite/issues/866 talks about "Configurable products shown in virtual category as discounted products, when they are not" but here the issue is that configurable products which have children with discounts are non showing at all. It's completely the opposite problem!!
Specifically I have a configurable product with 3 children with special prices and it's not showing in the discounted list!
Can somebody help please? Should we reopen the issue?

**Comment by Bashev** [comment]
@cptX, yes your issue can be solved by this improvement.

If you are using 2.10.14 or 2.11.x you need to enable indexing of child prices from the admin panel - `Store - Configuration - Elasticsuite - Catalog Search - Enable indexing discount on child products`

Run `php bin/magento indexer:reindex`

Enjoy.

**Comment by cptX** [comment]
Hi @Bashev , thanks for the quick reply. I was missing this config "Store - Configuration - Elasticsuite - Catalog Search - Enable indexing discount on child products".
Is this introduced with the patch #866?
I see a difference in the behavior now. Now I get many products that have no discount!
The only thing I have done is replaced the file PriceData.php with the one here https://github.com/Smile-SA/elasticsuite/pull/2677/files#diff-66291781f9e42ebe41f37807eec760ecc5d82c2e17eaa5665c09cda670fc3b7f

**Comment by Bashev** [comment]
If you have 2.11.0 you not need to replace anything, this PR is merged in the mentioned version. I'm confused why you are not seeing this setting and also you need to replace the file.

This setting and changes will reflect to your store in this case:
_ - If you have children (of configurable, grouped or bundle products) which are discounted by rule or by special price. This will set discounted `true` to the parent product. Which will be eligible to pass the rule "Discounted products only"._

NB: If you are migrated old store, or for some different reason you have special price on the configurable products (this happens most often if you migrate data from magento 1 - there configurable products have prices) whey will stay as discounted. You need to clear your database from prices for configurable products.

Check db - `catalog_product_entity_decimal` do you have entries for price and special_price attributes.

**Comment by cptX** [comment]
Things are simple! Sorry for the confusion. I just verified it worked!
Let's take things from the beginning.
Initially I was not aware of this option "Store - Configuration - Elasticsuite - Catalog Search - Enable indexing discount on child products". So after reading the threads I thought I need to replace the file PriceData.php with the new one.
By doing so and without selecting the option I didn't see any difference. Then you told me to set the option in configuration but I had already replaced the file so I saw many items that had no discount listed.
Now I replaced the file with the original with the option checked and finally worked!!
So conclusion is that in 2.11.0 the system was correctly functioning but without selecting this option doesn't work as expected!
Thank you all guys!
